images/operator-sdk/Dockerfile

```diff
@@ -17,18 +17,7 @@ COPY . .
 RUN GOOS=linux GOARCH=$TARGETARCH make build/operator-sdk
 
 # Final image.
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.9
-
-ENV GO_VERSION 1.19
-
-ARG TARGETARCH
-RUN microdnf install -y make gcc which tar gzip
-RUN curl -sSLo /tmp/go.tar.gz https://golang.org/dl/go${GO_VERSION}.linux-${TARGETARCH}.tar.gz \
-	&& rm -rf /usr/local/go \
-	&& tar -C /usr/local -xzf /tmp/go.tar.gz \
-	&& ln -sf /usr/local/go/bin/* /usr/local/bin/ \
-	&& rm -f /tmp/go.tar.gz \
-	&& go version
```

Comment:
Could you elaborate on why this was removed?

I believe that Go is required to be present in the final image due to the go plugin running things like `go mod tidy` at the end of scaffolding of files. Without Go being present in the final image, using this image will fail to scaffold Go based operator projects.

I believe that the ubi micro images don't include a package manager so we might be able to `COPY --from=builder ...` the necessary stuff to have Go "installed" in the final image.

Reply:
That should be removed from the runtime not the builder image. `go mod tidy` should be run at build time in the builder image. Is that not the case?
If I understand you correctly you are saying that the same image is used for build and run time. If that's the case it should get split for introducing UBI-micro. Adding golang, tar would kill the purpose (reducing the attack surface) of the exercise.

Reply:
>go mod tidy should be run at build time in the builder image. Is that not the case?

The `go mod tidy` operation that I am referring to is run as part of the scaffolding logic that gets run when you are creating a new Go based operator project using `operator-sdk init ...`

Reply:
I am probably missing the target. For golang operator there is [in kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v4/scaffolds/internal/templates/dockerfile.go#L68):
`FROM gcr.io/distroless/static:nonroot`
this gets replaced in operator-sdk so that `operator-sdk init` generates (without removing the comment about distroless :-) )
`FROM registry.access.redhat.com/ubi8/ubi-minimal:8.8`
This is where I would like to see ubi-micro.

Reply:
That replacement should only occur in an operator scaffolded with the OpenShift version of the operator-sdk and not the community (operator-framework) version of the operator-sdk.